docs/source/stability_policy.rst

```diff
@@ -0,0 +1,129 @@
+   This is incremented when new functionality or backwards compatible changes are introduced by PTB.
+   *This is also incremented when PTB adds support for a new Bot API version, which may include backwards incompatible changes in some cases as outlined* :ref:`below <bot-api-versioning>`.
+-  Z is the patch version.
+   This is incremented if backwards compatible bug fixes or smaller changes are introduced.
+
+Deprecation
+~~~~~~~~~~~
+
+From time to time we will want to change the behavior of an API or remove it entirely, or we do so to comply with changes in the Telegram Bot API.
+In those cases, we follow a deprecation schedule as detailed below.
+
+Functionality is marked as deprecated by a corresponding note in the release notes and the documentation.
+Where possible, a :class:`~telegram.warnings.PTBDeprecationWarning` is issued when deprecated functionality is used, but this is not mandatory.
+
+From time to time, we may decide to deprecate an API that is particularly widely used.
+In these cases, we may decide to provide an extended deprecation period, at our discretion.
+
+With version 20.0.0, PTB introduced majorly structural breaking changes without the above deprecation period.
+Should a similarly big change ever be deemed necessary by the development team and should a deprecation period prove too much additional effort, this violation of the stability policy will be announced well ahead of the release in our channel.
+
+Non-Bot API Functionality
+#########################
+
+Starting with version NEXT.VERSION, deprecated functionality will stay available for the current and the next major version.
+More precisely:
+
+-  In PTB X.Y.Z the feature exists
+-  In PTB X.Y.(Z + 1) or X.(Y + 1).* the feature is marked as deprecated
+-  In PTB (X + 1).*.* the feature is marked as deprecated
+-  In PTB (X + 2).0.0 the feature is removed or changed
+
+.. _bot-api-versioning:
+
+Bot API Functionality
+#####################
+
+As PTB has no control over deprecations introduced by Telegram and the schedule of these deprecations rarely coincides with PTBs deprecation schedule, we have a special policy for Bot API functionality.
+
+Starting with NEXT.VERSION, deprecated Bot API functionality will stay available for the current and the next major version *or* until the next version of the Bot API.
+More precisely, two cases are possible.
+
+Case 1
+^^^^^^
+
+-  In PTB X.Y.Z the feature exists
+-  Bot API version N.M is released and deprecates the feature
+-  In PTB X.(Y + 1).0 adds support for Bot API A.B and the feature is
+   marked as deprecated
+-  In PTB (X + 1).0.0 the feature is removed or changed
+
+Case 2
+^^^^^^
+
+-  In PTB X.Y.Z the feature exists
+-  Bot API version N.M is released and deprecates the feature
+-  PTB X.(Y + 1).0 adds support for Bot API N.M and the feature is marked as deprecated
+-  Bot API version N.(M + 1) is released
+-  In PTB X.(Y + 2).0 adds support for Bot API N.(M + 1) and the feature is removed or changed
```

Comment:
```suggestion
-  Bot API version N.(M + k) is released
-  In PTB X.(Y + i).0 adds support for Bot API N.(M + 1) and the feature is removed or changed
```

it doesn't have to be consecutive does it? Now this is getting really complicated...